src/pkcs11/framework-pkcs15.c

```diff
@@ -1070,9 +1070,13 @@ pkcs15_init_slot(struct sc_pkcs15_card *p15card, struct sc_pkcs11_slot *slot,
 			pin_info = NULL;
 		}
 		else   {
-			if (auth->label[0] && strncmp(auth->label, "PIN", 4) != 0)
-				snprintf(label, sizeof(label), "%.*s (%s)", (int) sizeof auth->label, auth->label, p15card->tokeninfo->label);
-			else
+			if (auth->label[0] && strncmp(auth->label, "PIN", 4) != 0) {
+				int ti_len = MIN(strlen(p15card->tokeninfo->label),
+					sizeof(label) - strlen(auth->label) - 3);
+				snprintf(label, sizeof(label), "%.*s (%.*s)",
+					(int) strlen(auth->label), auth->label,
```

Comment:
this looks stupid. shouldn't it be `(int) MIN(sizeof auth->label, strlen(auth->label)`? `ti_len` should be calculated accordingly.

Reply:
This is using assumption, that `auth->label` as a first argument should not be too long and therefore fit (though it is not good assumption). I am not sure what your `MIN` should give us, since the `label` is static buffer `char label[SC_PKCS15_MAX_LABEL_SIZE];`, therefore `sizeof()` is always 255, isn't it?
Fitting two flexible-length characters into fixed buffer is this way complicated, but the attempt here is to avoid truncation in the end of character, but keep the second buffer contained neatly in the brackets.